assets/blocks/quiz/quiz-block/quiz-settings-answer-feedback.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * WordPress dependencies
+ */
+import { BaseControl, ToggleControl } from '@wordpress/components';
+import { useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Answer feedback settings for the quiz block.
+ *
+ * @param {Object}   props
+ * @param {Object}   props.options       Quiz options.
+ * @param {Function} props.setAttributes
+ */
+const QuizAnswerFeedbackSettings = ( { options, setAttributes } ) => {
+	let {
+		failedShowAnswerFeedback = null,
+		failedShowCorrectAnswers = null,
+		failedIndicateIncorrect = null,
+		allowRetakes,
+	} = options;
+
+	const hasAnyOptionSet =
+		null !== failedShowAnswerFeedback ||
+		null !== failedShowCorrectAnswers ||
+		null !== failedIndicateIncorrect;
+
+	failedShowAnswerFeedback = failedShowAnswerFeedback ?? ! allowRetakes;
+	failedShowCorrectAnswers = failedShowCorrectAnswers ?? ! allowRetakes;
+	failedIndicateIncorrect = failedIndicateIncorrect ?? ! allowRetakes;
+
+	/**
+	 * Set default values based on 'Allow Retakes' if nothing was set.
+	 */
+	useEffect( () => {
+		if ( ! hasAnyOptionSet ) {
```

Comment:
With the current approach, I noticed that if we just check the "Pass Required" in a new lesson and save, it will update other values to different values. I wonder if it would be better to have defaults there regardless of the Allow Retakes. Or does it really need to be dependent for some reason?

Reply:
Oh looks like that was a different issue with the default options, fixing in 6038154.